sklearn/metrics/tests/test_classification.py

```diff
@@ -2337,6 +2337,79 @@ def test_brier_score_loss():
+
+    assert_almost_equal(multiclass_brier_score_loss(y_true, y_pred),
+                        .05083333)
+    assert_almost_equal(multiclass_brier_score_loss(y_true, y_pred),
+                        brier_score_loss(y_true, y_pred) * 2)
+
+    with pytest.raises(ValueError):
+        # bad length of y_pred
+        multiclass_brier_score_loss(y_true, y_pred[1:])
+    with pytest.raises(ValueError):
+        # y_pred has value greater than 1
+        multiclass_brier_score_loss(y_true, y_pred + 1.)
+    with pytest.raises(ValueError):
+        # y_pred has value less than 1
+        multiclass_brier_score_loss(y_true, y_pred - 1.)
+
+    # ensure to raise an error for wrong number of classes
+    y_true = np.array([0, 1, 2, 0])
+    y_pred = np.array([0.8, 0.6, 0.4, 0.2])
+    error_message = ("y_true and y_prob contain different number of "
+                     "classes 3, 2. Please provide the true "
+                     "labels explicitly through the labels argument. "
+                     "Classes found in "
+                     "y_true: [0 1 2]")
+    with pytest.raises(ValueError, match=re.escape(error_message)):
+        multiclass_brier_score_loss(y_true, y_pred)
+
+    y_true = ['eggs', 'spam', 'ham']
+    y_pred = [[1, 0, 0],
+              [0, 1, 0],
+              [0, 1, 0]]
+    labels = ['eggs', 'spam', 'ham', 'yams']
+    error_message = ("The number of classes in labels is different "
+                     "from that in y_prob. Classes found in "
+                     "labels: ['eggs' 'ham' 'spam' 'yams']")
+    with pytest.raises(ValueError, match=re.escape(error_message)):
+        multiclass_brier_score_loss(y_true, y_pred, labels=labels)
+
+    # raise error message when there's only one class in y_true
+    y_true = ['eggs']
+    y_pred = [.1]
+    error_message = (f'y_true contains only one label: {y_true[0]}. Please '
+                     f'provide the true labels explicitly through the '
+                     f'labels argument.')
+    with pytest.raises(ValueError, match=re.escape(error_message)):
+        multiclass_brier_score_loss(y_true, y_pred)
+
+    # error is fixed when labels is specified
+    assert_almost_equal(multiclass_brier_score_loss(y_true, y_pred,
+                                                    labels=['eggs', 'ham']),
+                        .02)
+
+    # test cases for multi-class
+    assert_almost_equal(
+        multiclass_brier_score_loss(['eggs', 'spam', 'ham'],
+                                    [[1, 0, 0, 0],
+                                     [0, 1, 0, 0],
+                                     [0, 1, 0, 0]],
+                                    labels=['eggs', 'spam', 'ham', 'yams']),
+        2/3)
+
+    assert_almost_equal(
+        multiclass_brier_score_loss([1, 0, 2],
+                                    [[0.2, 0.7, 0.1],
+                                     [0.6, 0.2, 0.2],
+                                     [0.6, 0.1, 0.3]]),
+        .41333333)
```

Comment:
Could you please split this test in 2 tests, one for invalid input and error messages and the other for expected numerical values of valid inputs?

Also could you please add a test that check that perfect predictions lead to 0 Brier and perfectly wrong prediction lead to a Brier of 2. (both for a 2-class and a 3-class classification problem).

Reply:
👍